src/test/java/uk/gov/companieshouse/insolvency/delta/transformer/InsolvencyApiTransformerTest.java

```diff
@@ -1,18 +1,58 @@
 package uk.gov.companieshouse.insolvency.delta.transformer;
 
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
-import uk.gov.companieshouse.api.delta.InsolvencyDelta;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.companieshouse.api.delta.Insolvency;
+import uk.gov.companieshouse.api.insolvency.InternalCompanyInsolvency;
+import uk.gov.companieshouse.insolvency.delta.mapper.InsolvencyMapper;
 
-import static org.assertj.core.api.Assertions.assertThat;
+@ExtendWith(MockitoExtension.class)
+class InsolvencyApiTransformerTest {
 
-public class InsolvencyApiTransformerTest {
+    private static final String COMPANY_NUMBER = "01876743";
 
-    private final InsolvencyApiTransformer transformer = new InsolvencyApiTransformer();
+    @Mock
+    private InsolvencyMapper mapper;
+
+    private InsolvencyApiTransformer transformer;
+
+    @BeforeEach
+    public void setup() {
+        transformer = new InsolvencyApiTransformer(mapper);
+    }
 
     @Test
-    public void transformSuccessfully() {
-        final InsolvencyDelta input = new InsolvencyDelta();
-        assertThat(transformer.transform(input)).isEqualTo(input.toString());
+    @DisplayName("Successfully invokes the mapstruct mapper")
+    void When_TransformerIsCalled_ThenMapperIsSuccessfullyInvoked() {
+        Insolvency insolvency = new Insolvency();
+        insolvency.setCompanyNumber(COMPANY_NUMBER);
+        InternalCompanyInsolvency mockInternalInsolvency = mock(InternalCompanyInsolvency.class);
+
+        when(mapper.insolvencyDeltaToApi(insolvency, COMPANY_NUMBER))
+                .thenReturn(mockInternalInsolvency);
+
+        InternalCompanyInsolvency actual = transformer.transform(insolvency);
+        assertThat(actual).isEqualTo(mockInternalInsolvency);
     }
 
+    @Test
+    @DisplayName("Throws an error when there's an issue with the transformation")
+    void When_ErrorDuringTransformation_ThenThrows() {
```

Comment:
just a suggestion, maybe better to keep test method naming consistent with camelcase sentence instead of underscores , for example this test can be named "testInsolvencyTransformationWithInvalidDataThrowsError" maybe?

Reply:
We were following the same naming conventions in the other tests to be consistent which also follows Java unit test naming conventions. Think it also makes the tests easier to read which also follows BDD syntax of given/when/then.

In the future, we will make an effort to add the display name annotation to each unit test - it should be clear enough and that should be our focus, as it is more human readable than any function name.